### PR TITLE
Partial fix for #6755: create more python type hints files 53 to 56

### PR DESCRIPTION
**Pull request by scottbarnes**
Partial fix for #6755

refactor

### Stakeholders

@cclauss

### CODE REVIEW

File: scripts/solr_builder/solr_builder/solr_builder.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import json
```
Comment (cclauss) [inline scripts/solr_builder/solr_builder/solr_builder.py:1]:
```suggestion
from __future__ import annotations

import json
```
This might magically fix our failing GitHub Action.

File: scripts/solr_builder/solr_builder/solr_builder.py [hunk 2]
```diff
@@ -107,7 +105,13 @@ def query_iter(self, query, size=20):
 
         cur.close()
 
-    def query_batched(self, query, size, cursor_name=None, cache_json=False):
+    def query_batched(
+        self,
+        query: str,
+        size: int,
+        cursor_name: Optional[str] = None,
```
Comment (cclauss) [inline scripts/solr_builder/solr_builder/solr_builder.py:112]:
```suggestion
        cursor_name: str = None,
```
For function and method parameters with a default value of `None` we can remove the `Optional[]` because both Python and mypy are smart enough to figure this out.  The hint is telling the caller to provide a `str` for this parameter but it is OK it the parameter is not provided.  Doing this greatly decreases the visual clutter.

**Review by cclauss — CHANGES_REQUESTED** [review]
This is great.  Ping on Slack if these comments are unclear.

File: scripts/solr_builder/solr_builder/solr_builder.py (hunk 1 above)
Comment (cclauss) [reply scripts/solr_builder/solr_builder/solr_builder.py:1]: I will add this `__future__` import and then (as we discussed on Slack) look at the [commits](https://github.com/internetarchive/openlibrary/pull/6826/commits) to see all the changes that `pre-commit` + `pyupgrade` make based on it’s presence.  Does that automatic commit fix our failing test?

Remember to `git pull` these changes to your local branch before making any further changes.

**Comment by cclauss** [comment]
I added the `__future__` import which fixed our failing test.

**Remember to `git pull`** the [changes](https://github.com/internetarchive/openlibrary/pull/6826/commits) from `pre-commit`’s and my commits to your local branch before making any further changes.

File: scripts/solr_builder/solr_builder/solr_builder.py [hunk 3]
```diff
@@ -399,17 +405,17 @@ def log(self, entry):
...
-        ):
+            seen: str | int = None,
+            total: str | int = None,
+            percent: str | float = None,
+            elapsed: str | float = None,
+            q_1: str | float = None,
+            q_auth: str | float = None,
+            cached: str | int = None,
+            q_ia: str | float = None,
+            ia_cache: str | int = None,
```
Comment (cclauss) [inline scripts/solr_builder/solr_builder/solr_builder.py:416]: Does this work?
```suggestion
            seen: str | int | None = None,
            total: str | int | None = None,
            percent: str | float | None = None,
            elapsed: str | float | None = None,
            q_1: str | float | None = None,
            q_auth: str | float | None = None,
            cached: str | int | None = None,
            q_ia: str | float = None,
            ia_cache: str | int | None = None,
```

File: scripts/solr_updater.py [hunk 4]
```diff
@@ -30,7 +30,7 @@
 args: dict = {}
 
 
-def read_state_file(path, initial_state: str = None):
+def read_state_file(path, initial_state: Optional[str] = None):
```
Comment (cclauss) [inline scripts/solr_updater.py:33]: Why this change?

File: scripts/solr_updater.py [hunk 5]
```diff
@@ -45,7 +45,7 @@ def get_default_offset():
 
 
 class InfobaseLog:
-    def __init__(self, hostname, exclude=None):
+    def __init__(self, hostname: str, exclude: Optional[str] = None):
```
Comment (cclauss) [inline scripts/solr_updater.py:48]: Why this change?

File: scripts/tests/test_copydocs.py [hunk 6]
```diff
@@ -20,14 +21,14 @@ def test_to_uri(self):
...
+    def __init__(self, docs: list[dict]):
         """
         :param list[dict] docs:
         """
-        self.db = {}  # Mapping of key to (Mp of revision to doc)
+        self.db: dict = {}  # Mapping of key to (Mp of revision to doc)
         self.save_many(docs)
 
-    def get(self, key, revision=None):
+    def get(self, key: str, revision: Optional[int] = None) -> Optional[dict]:
```
Comment (cclauss) [inline scripts/tests/test_copydocs.py:31]:
```suggestion
    def get(self, key: str, revision: int = None) -> dict | None:
```

File: scripts/solr_builder/solr_builder/solr_builder.py (hunk 3 above)
Comment (scottbarnes) [reply scripts/solr_builder/solr_builder/solr_builder.py]: Yes. Thanks!

File: scripts/solr_updater.py (hunk 5 above)
Comment (scottbarnes) [reply scripts/solr_updater.py]: This was an older change, since undone, that I may have inadvertently re-pushed.

File: scripts/solr_updater.py (hunk 4 above)
Comment (scottbarnes) [reply scripts/solr_updater.py]: This was an older change, since undone, that I may have inadvertently re-pushed.

**Review by cclauss — APPROVED** [review]
Nice!  Thanks for doing this.
